### PR TITLE
API-26681/ DocServer & LOAST: Upgrade typescript version 

### PR DESCRIPTION
**Pull request by mheusgit**
- Updated typescript to v5.1.3
- Jest updated to 29.5.0 & ts-jest to 29.1.0.  Updated jest.config.js to follow new convention
- Minor TS issue corrected in apiVersion.service.ts
- Jest team is fixing issue bypassed loast.manager.test.ts

### CODE REVIEW

**Comment by mheusgit** [comment]
Jenkins build issue is tied to LOAST version.  PR cannot be merged until after LOAST changes.
https://github.com/department-of-veterans-affairs/lighthouse-oas-tests/pull/134

File: src/api/version/apiVersion.service.ts [hunk 1]
```diff
@@ -183,7 +184,7 @@ class APIVersionService {
     try {
       return this.entityManager.find(
         APIVersionModel,
-        { openAPISpecs: { status } },
+        { openAPISpecs: { status: status as unknown as StatusEnum } },
```
Comment (dsteeve) [inline src/api/version/apiVersion.service.ts:187]: not loving this "unknown" cast at all, but see the lint error without it.  :-(.

**Review by dsteeve — DISMISSED** [review]
pointed to the local build of LOAST and was able to build this and run the tests cleanly.

File: src/api/version/apiVersion.service.ts (hunk 1 above)
Comment (mheusgit) [reply src/api/version/apiVersion.service.ts:187]: Yeah.  Not sure what change caused this line to start failing linting.  Maybe a minor correction on eslint's part?

**Review by dsteeve — APPROVED** [review]
looks good now

**Pull request merged**